infrastructure/comms/src/connection/net_address/ip.rs

```diff
@@ -25,6 +25,7 @@ use std::{net::SocketAddr, str::FromStr};
 use crate::connection::NetAddressError;
 
 /// Represents an {IPv4, IPv6} address and port
+#[derive(Clone, Hash, PartialEq, Eq, Debug)]
```

Comment:
again do we need all of these?

Reply:
`Hash` will go - the rest I think are fine, except for Debug? I imagine that having a debug output for an address could be useful - I'm never sure `Debug` these should be added almost by default or added just to debug and removed again once done